server/executor/poller_executor.go

```diff
@@ -115,8 +114,7 @@ func (pe DefaultPollerExecutor) donePollingTraces(job *PollingRequest, trace tra
 		return false
 	}
 
-	// We always expect to get the tracetest trigger span, so it has to have more than 1 span
```

Comment:
This is only true if the user configures tracetest to send the triggering span to the same data store as its application. So, to prevent infinite loops, I reverted the `len() > 1` condition-